src/screens/SetupScreen/index.tsx

```diff
@@ -80,13 +78,41 @@ export class Setup extends React.Component<Props, SetupScreenState> {
     }
 
     public componentWillReceiveProps(nextProps: Props) {
+        if (!this.props.user.email && nextProps.user.email && this.state.currentStep === '1') {
+            this.setState({
+                currentStep: '2',
+            });
+        }
+
+        if (!this.props.platformCreateSuccess && nextProps.platformCreateSuccess) {
+            this.setState({
+                currentStep: '3',
+            });
+        }
+
         if (!this.props.marketsSuccess && nextProps.marketsSuccess) {
             this.setState({
                 currentStep: '4',
             });
         }
     }
 
+    public componentDidMount() {
+        if (wizardStep() !== '1') {
+            this.setState({
+                currentStep: wizardStep(),
+            });
+        }
+    }
+
+    public UNSAFE_componentWillReceiveProps(nextProps: Props) {
```

Comment:
there is two componentWillReceiveProps in this file. Combine them into one and use componentDidUpdate